### PR TITLE
[DYNAREC] Revert some fasly changes to the shift opcodes

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

File: src/dynarec/arm64/dynarec_arm64_67.c [hunk 1]
```diff
@@ -973,51 +957,31 @@ uintptr_t dynarec64_67(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
...
-                        FAKEED;
-                    F8;
+                    SETFLAGS(X_ALL, SF_SET_PENDING);    // some flags are left undefined
+                    GETED32(1);
+                    u8 = (F8)&(rex.w?0x3f:0x1f);
                     emit_sar32c(dyn, ninst, rex, ed, u8, x3, x4);
-                    if (u8) {
+                    if(u8) {
                         WBACK;
-                    } else if (MODREG && !rex.w)
```
Comment (ptitSeb) [inline src/dynarec/arm64/dynarec_arm64_67.c:1019]: why removing this part?

Comment (ptitSeb) [reply src/dynarec/arm64/dynarec_arm64_67.c:1019]: Ah, it was not there before, I see

Comment (ksco) [reply src/dynarec/arm64/dynarec_arm64_67.c:1019]: Yeah, I reverted the changes made to arm64.

**Comment by ksco** [comment]
Here for a cleaner review:
https://github.com/ptitSeb/box64/compare/55c2c9ee59261540714c3861875d38d66e22ac37...ksco:box64:shift

**Pull request merged**